pkg/config/polling_manager.go

```diff
@@ -205,7 +186,40 @@ func NewPollingProjectConfigManager(sdkKey string, pollingMangerOptions ...Optio
 	}
 
 	initDatafile := pollingProjectConfigManager.initDatafile
-	pollingProjectConfigManager.SyncConfig(initDatafile) // initial poll
+	if len(initDatafile) == 0 {
+		pollingProjectConfigManager.notificationCenter = registry.GetNotificationCenter(sdkKey)
+		pollingProjectConfigManager.SyncConfig() // initial poll
+	} else {
+		if projectConfig, _ := datafileprojectconfig.NewDatafileProjectConfig(initDatafile); projectConfig != nil {
+			_ = pollingProjectConfigManager.setConfig(projectConfig)
+		}
+		pollingProjectConfigManager.notificationCenter = registry.GetNotificationCenter(sdkKey)
```

Comment:
Please add comment, notificationcenter is set later to avoid hard-coded config update notification.